### PR TITLE
The big merge

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Closes #100 
Closes #103 
Closes #105 
Closes #110 
Closes #112 
Closes #117 

TL;DR: Let's merge this big PR instead of the each of ones above.

Yesterday, we needed to test all these PRs at once, because we want:

1. partner repacks to ship as soon as possible
1. signed addons to be shipped this week
1. Thunderbird workers to not be on one's puppet env.

Merging these 6 PRs altogether created some conflicts which, I think I resolved. The latest staging release shows no beetmover error except chain of trust ones: https://tools.taskcluster.net/groups/fzAr-EKyTY6JY4FqPcgbFA.

I think it's faster, safer, and thriftier if I land this PR as is on master, instead of landing the small ones. If we went the small PR way, we'd have to wait for everybody to rebase their patches on top of whatever landed on master. This may introduce errors which are (supposedly) solved here. 

@escapewindow, @tomprince, @Callek : What do you all think?

### CODE REVIEW

**Comment by MihaiTabara** [comment]
Thank you @JohanLorenzo, this is amazing!
